rclcpp/test/rclcpp/test_node.cpp

```diff
@@ -2798,9 +2804,8 @@ TEST_F(TestNode, static_and_dynamic_typing) {
     EXPECT_EQ("hello!", param);
   }
   {
-    EXPECT_THROW(
-      node->declare_parameter("integer_override_not_given", rclcpp::PARAMETER_INTEGER),
-      rclcpp::exceptions::NoParameterOverrideProvided);
+    auto param = node->declare_parameter("integer_override_not_given", rclcpp::PARAMETER_INTEGER);
+    EXPECT_EQ(rclcpp::PARAMETER_NOT_SET, param.get_type());
```

Comment:
is this any different to the test case added in L342?

Reply:
No, I can remove the case I added and leave this one

Reply:
I don't mind much, but I think we should remove one of them to avoid duplication.

Reply:
Removed duplicate test in 01e5ae2